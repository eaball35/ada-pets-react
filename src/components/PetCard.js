import React, { Component } from 'react';
import PropTypes from 'prop-types';

import 'bootstrap/dist/css/bootstrap.min.css';
import './PetCard.css';

import speciesEmoji from '../speciesEmoji';

class PetCard extends Component {
  constructor(props) {
    super();
  }

  render() {
    return (
      <div className="card pet-card">

        <section className="pet-card--header">

          {speciesEmoji(this.props.species)} {this.props.id} - {this.props.name}
          <button
            className="btn btn-primary pet-card--select-pet-btn"
            onClick={this.props.onSelectPet}
          >
            Select
          </button>
          <button
            type="button"
            className="btn btn-danger pet-card--remove-btn"
            aria-label="Remove"
            onClick={this.props.onRemovePet}
          >
            Remove
          </button>
        </section>
        <section className="pet-card--body">
          {this.props.about.length > 128 ? `${ this.props.about.substring(0, 128) }...` : this.props.about}
        </section>
        <section className="pet-card--footer text-muted">
          {this.props.location}
        </section>
      </div>
    );
  };
};

PetCard.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired,
  about: PropTypes.string,
  location: PropTypes.string,
  onRemovePet: PropTypes.func.isRequired,
  onSelectPet: PropTypes.func.isRequired,
}

export default PetCard;