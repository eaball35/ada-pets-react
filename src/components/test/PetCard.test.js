import React from 'react';
import { render, cleanup } from '@testing-library/react'
import PetCard from '../PetCard';

const petCard = <PetCard
  id= ''
  name='Denali'
  species=''
  about=''
  location='Seattle, WA'
  onSelectPet={() => { }}
  onRemovePet={() => { }}
/>

describe('NewPetForm', () => {  
  test('that it matches the existing snapshot', () => {
    // Arrange-Act
    const { asFragment } = render(
      petCard
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });

  test('that it shows the dogs name & location', () => {
    // Arrange-Act
    const result = render(
      petCard
    );
    // Assert
    expect(result.getByText(/Denali/)).toBeDefined();
    expect(result.getByText(/Seattle, WA/)).toBeDefined();
    // Add test for location display
  });


  // src/components/test/PetCard.test.js
  test('The "selectPetCallback" function is called when the `select` button is clicked on', () => {

    // Arrange
    // Create a mock callback function
    const selectPet = jest.fn();

    // Render a PetCard
    const container = render(<PetCard
      id={1}
      name={"Samson"}
      species={"Cat"}
      about={"A very awesome cat!  Dont' touch the hair"}
      location={"Seattle, WA"}
      onRemovePet={() => { }}
      onSelectPet={selectPet}
    />);

    // Act
    // Find the "Select" button
    const selectButton = container.getByText(/Select/);
    // Trigger a 'click' event
    selectButton.click();

    // Assert
    // Verify that the callback function was called.
    expect(selectPet).toHaveBeenCalled();
  });

  test('The "removePetCallback" function is called when the `remove` button is clicked on', () => {

    // Arrange
    // Create a mock callback function
    const removePet = jest.fn();

    // Render a PetCard
    const container = render(<PetCard
      id={1}
      name={"Samson"}
      species={"Cat"}
      about={"A very awesome cat!  Dont' touch the hair"}
      location={"Seattle, WA"}
      onRemovePet={removePet}
      onSelectPet={() => { }}
    />);

    // Act
    // Find the "Select" button
    const selectButton = container.getByText(/Remove/);
    // Trigger a 'click' event
    selectButton.click();

    // Assert
    // Verify that the callback function was called.
    expect(removePet).toHaveBeenCalled();
  });
  // afterEach(cleanup());
});