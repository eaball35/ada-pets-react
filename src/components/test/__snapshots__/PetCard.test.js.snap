// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPetForm that it matches the existing snapshot 1`] = `
<DocumentFragment>
  <div
    class="card pet-card"
  >
    <section
      class="pet-card--header"
    >
      ‚ùì  - Denali
      <button
        class="btn btn-primary pet-card--select-pet-btn"
      >
        Select
      </button>
      <button
        aria-label="Remove"
        class="btn btn-danger pet-card--remove-btn"
        type="button"
      >
        Remove
      </button>
    </section>
    <section
      class="pet-card--body"
    />
    <section
      class="pet-card--footer text-muted"
    >
      Seattle, WA
    </section>
  </div>
</DocumentFragment>
`;
